### PR TITLE
feat: display model thoughts

### LINKED ISSUE #5159: [Multi Actions] handle cases where we have content and calls functions in the same message
It looks a bit like a gpt4 bug.
In this case, we likely want to ignore the generation.

Here's an example run where it happened: https://dust.tt/w/78bda07b39/a/0e9889c787/runs/3243be92381110215723b48a155aa1a6d8f7ea3790d38acb322a3d52bc0b50ac

Right now, we handle it poorly since we show the generation being streamed, and then disappear when the function call message is complete, and then start streaming again with the "real" generation.

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Fixes https://github.com/dust-tt/dust/issues/5159

Screenshot 2024-06-12 at 10 00 01

Now that we stream and store Chain Of Thoughts, this PR also renders it in the UI, using a distinct "Model thoughts" block in which the chain of thoughts accumulate.
I also added a `swallow` boolean to model delimiter configuration entries that control whether the content in between that type of delimiters should be "swallowed". That allows to hide the "search_quality_score" from Anthropic which is not useful and not straightforward to "productize".

## Risk

Critical path

## Deploy Plan

N/A

### CODE REVIEW

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 1]
```diff
@@ -429,24 +476,52 @@ export function AgentMessage({
     }
 
     // TODO(2024-05-27 flav) Use <ConversationMessage.citations />.
+
+    const chainOfThought = agentMessage.chainOfThoughts.join("");
```
Comment (fontanierh) [inline front/components/assistant/conversation/AgentMessage.tsx:480]: For now, we render all the chain of thoughts in one box. It's simpler (no need to detect while streaming that it's a "new" CoT) and rendering multiple CoTs needs some design IMO

**Review by PopDaph — APPROVED** [review]
I got partial context on this since I did not review the former PRs and did not take the time to look at them, but LGTM 👍🏻

**Review by spolu — APPROVED** [review]

**Comment by spolu** [comment]
(assuming testing thorough of legacy agents before merging)

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 2]
```diff
@@ -182,33 +190,63 @@ export function AgentMessage({
...
+            setStreamedAgentMessage((m) => {
+              const previousContent = m.content || "";
+              return { ...m, content: previousContent + event.text };
+            });
+            break;
+          case "chain_of_thought":
+            setLastTokenClassification("chain_of_thought");
+            setStreamedAgentMessage((m) => {
+              const currentChainOfThoughts = m.chainOfThoughts;
+              const lastChainOfThought = currentChainOfThoughts.pop() ?? "";
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant/conversation/AgentMessage.tsx:234]: OP here should help provide some context: https://github.com/dust-tt/dust/pull/5568

File: front/components/assistant/conversation/AgentMessage.tsx [hunk 3]
```diff
@@ -107,6 +110,10 @@ export function AgentMessage({
     }
   })();
 
+  const [lastTokenClassification, setLastTokenClassification] = useState<
+    null | "tokens" | "chain_of_thought"
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant/conversation/AgentMessage.tsx:114]: We really only use this particular enum here (other places use a broader enum that include "opening_delimiter" and "closing_delimiter")

**Comment by fontanierh** [comment]
> (assuming testing thorough of legacy agents before merging)

Added support for legacy mode @spolu  👍

File: front/lib/api/assistant/generation.ts [hunk 4]
```diff
@@ -493,20 +497,40 @@ export async function* runGeneration(
...
+  | AgentChainOfThoughtEvent,
   void
 > {
   const owner = auth.workspace();
   if (!owner) {
     throw new Error("Unexpected unauthenticated call to `runGeneration`");
   }
 
-  const { model } = configuration;
+  const model = SUPPORTED_MODEL_CONFIGS.find(
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:513]: isn't that `getSupportedModelConfig` ?

**Review by spolu — APPROVED** [review]

**Comment by spolu** [comment]
Thanks change loooks good. Just left a small comment.

File: front/lib/api/assistant/generation.ts (hunk 4 above)
Comment (fontanierh) [reply front/lib/api/assistant/generation.ts:513]: Didn't know. happy to change it 🙏

Comment (spolu) [reply front/lib/api/assistant/generation.ts:513]: You touched it a few lines below? :p

**Pull request merged**